kuma/search/serializers.py

```diff
@@ -6,13 +6,20 @@
 
 
 class SearchQuerySerializer(serializers.Serializer):
-    q = serializers.CharField(required=False)
+    q = serializers.CharField(required=False, max_length=1024)
     highlight = serializers.BooleanField(required=False, default=True)
     # Advanced search query paramenters.
     css_classnames = serializers.CharField(required=False)
     html_attributes = serializers.CharField(required=False)
     kumascript_macros = serializers.CharField(required=False)
 
+    def validate_q(self, value):
+        # Check that \n not in query
+        if '\\n' in value.lower():
```

Comment:
```suggestion
        if '\n' in value.lower():
```
otherwise it's `slash-and-newline`.

Reply:
Also, you need not to `value.lower()` since `\n` will be there in either case.